Add v1.4.1 release notes

tserong opened the pull request.

innobead commented on v1.4.1.md:
```diff
@@ -0,0 +1,157 @@
+
+## Known Issues
+
+- [backport v1.4] [BUG] Fail to trigger v1.4.0 to v1.4.1-rc1 upgrade on witness cluster due to managed chart harvester is not ready [#7414](https://github.com/harvester/harvester/issues/7414)
+- [BUG] Fail to trigger v1.4.0 to v1.4.1-rc1 upgrade on witness cluster due to managed chart harvester is not ready [#7375](https://github.com/harvester/harvester/issues/7375)
```
Duplicated issues.

ibrokethecloud replied:
should we change the target to v1.4.1?

tserong replied:
Probably -- the text is just lifted from the issue title, but it might be nicer to just say v1.4.0 (without rc1) in the release notes

tserong commented:
I've removed the duplicate issue and also changed mentions of `v1.4.1-rc` to `v1.4.1` (I know the original bug titles still say `rc1`, but I agree it looks better in the release notes to drop that piece as @ibrokethecloud suggested)

jillian-maroket reviewed (approved):
LGTM. Thanks!

Vicente-Cheng reviewed (approved):
Thanks!!
